docs/src/main/docs/guides/05_Dockerfile.adoc

```diff
@@ -0,0 +1,302 @@
+
+Since the `target` directory has the runtime dependencies, the
+Dockerfile for our application is pretty simple. You can
+find the Dockerfile in the generated project at `src/main/docker/Dockerfile`.
+It should look like this:
+
+[source,yaml,subs="verbatim,attributes"]
+----
+FROM openjdk:8-jre-slim 
+
+RUN mkdir /app
+COPY libs /app/libs 
+COPY ${project.artifactId}.jar /app
+
+CMD ["java", "-jar", "/app/${project.artifactId}.jar"]
+----
+
+* We use a Java 8 JRE image provided by the OpenJDK project.
+* We then create a directory to hold our application and copy the `libs` directory
+  into it followed by the application jar. The command to start the application is
+  just like the one we used when running natively on our desktop. 
+
+What is `${project.artifactId}`? It's a Maven property. The project uses
+the `maven-resources-plugin` to filter the Dockerfile while copying it to the
+target directory. So this property gets expanded to the project artifactId
+(`quickstart-se` for example).
+
+Since the quickstart project already contains the Dockerfile you can go ahead and try it:
+
+[source,bash,subs="verbatim,attributes"]
+----
+docker build -t quickstart-se target
+
+Sending build context to Docker daemon  5.641MB
+Step 1/5 : FROM openjdk:8-jre-slim
+ ---> 3e85180d5f58
+Step 2/5 : RUN mkdir /app
+ ---> Using cache
+ ---> d24e2f320e6b
+Step 3/5 : COPY libs /app/libs
+ ---> Using cache <1>
+ ---> 9772d4c5d4a0
+Step 4/5 : COPY quickstart-se.jar /app
+ ---> f156df1d0338
+Step 5/5 : CMD ["java", "-jar", "/app/quickstart-se.jar"]
+ ---> Running in 29838194f452
+Removing intermediate container 29838194f452
+ ---> 6a634dbe3ecf
+Successfully built 6a634dbe3ecf
+Successfully tagged quickstart-se:latest
+----
+
+<1> The first time you run `docker build` you won't see `Using cache` for this
+    layer, but on subsequent builds you should. This is good. It
+    means that the image layer that contains our runtime dependencies is
+    not modified every time we build our application. Only the layer containing
+    the application jar is. That means if we're pushing our docker image to a
+    remote registry then the layer with our runtime dependencies does not 
+    need to be pushed every time we rebuild.
+
+You can now run the docker container:
+
+[source,bash,subs="verbatim,attributes"]
+----
+docker run --rm -p 8080:8080 quickstart-se:latest
+----
+
+[source,bash,subs="verbatim,attributes"]
+----
+curl -X GET http://localhost:8080/greet
+{"message":"Hello World!"}
+----
+
+=== Why no Fat Jar?
+
+Fat Jars are jar files that contain the application and all runtime
+dependencies. This is handy because it's one file with all you need
+to run your application.
+
+The problem with fat jars is that they are not optimal when used in
+a docker image. That's because the image layer that contains your
+application also contains all of its runtime dependencies, and that
+means more data to push to a docker registry every time you rebuild
+your application.
+
```

Comment:
Another problem (I think) with fat jars is that the content of any dependent JAR that was signed loses its "signed-ness" when extracted and merged into the fat JAR. This is probably far less important than fat JARs being sub-optimal for Docker. If we don't want to mention that other issue explicitly maybe change `The problem with fat jars` to `A problem...`?

Reply:
OK